FirebaseStorageSwift/Tests/Unit/StorageAPITests.swift

```diff
@@ -0,0 +1,210 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// MARK: This file is used to evaluate the experience of using Firebase APIs in Swift.
+
+import Foundation
+
+import FirebaseCore
+import FirebaseStorage
+
+final class StorageAPITests {
+  func StorageAPIs() {
+    let app = FirebaseApp.app()
+    _ = Storage.storage()
+    _ = Storage.storage(app: app!)
+    _ = Storage.storage(url: "my-url")
+    let storage = Storage.storage(app: app!, url: "my-url")
+    _ = storage.app
+    storage.maxUploadRetryTime = storage.maxUploadRetryTime
+    storage.maxDownloadRetryTime = storage.maxDownloadRetryTime + 1
+    storage.maxOperationRetryTime = storage.maxOperationRetryTime + 1
+    storage.callbackQueue = storage.callbackQueue
+    _ = storage.reference()
+    _ = storage.reference(forURL: "my-url")
+    _ = storage.reference(withPath: "path")
+    storage.useEmulator(withHost: "host", port: 123)
+  }
+
+  func StorageReferenceApis() {
+    var ref = Storage.storage().reference()
+    _ = ref.storage
+    _ = ref.bucket
+    _ = ref.fullPath
+    _ = ref.name
+
+    ref = ref.root()
+    ref = ref.parent()!
+    ref = ref.child("path")
+
+    _ = ref.putData(Data())
+    _ = ref.putData(Data(), metadata: nil)
```

Comment:
Can we add in a Metadata instance here and  below to confirm the type doesn't change, as well as one with `nil`?